nemo/collections/asr/models/msdd_models.py

```diff
@@ -0,0 +1,1382 @@
+
+    def diar_infer(
+        self, test_batch: List[torch.Tensor], test_data_collection: List[Any]
+    ) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+        """
+        Launch forward_infer() function by feeding the session-wise embedding sequences to get pairwise speaker prediction values.
+        If split_infer is True, the input audio clips are broken into short sequences then cluster average embeddings are calculated
+        for inference. Split-infer might result in an improved results if calculating clustering average on the shorter tim-espan can
+        help speaker assignment.
+
+        Args:
+            test_batch: (list)
+                List containing embedding sequences, length of embedding sequences, ground truth labels (if exists) and initializing embedding vectors.
+            test_data_collection: (list)
+                List containing test-set dataloader contents. test_data_collection includes wav file path, RTTM file path, clustered speaker indices.
+
+        Returns:
+            preds (Tensor):
+                Tensor containing predicted values which are generated from MSDD model.
+            targets (Tensor):
+                Tensor containing binary ground-truth values.
+            signal_lengths (Tensor):
+                The actual Session length (number of steps = number of base-scale segments) without zero padding.
+        """
+        signals, signal_lengths, _targets, emb_vectors = test_batch
+        if self._cfg.diarizer.msdd_model.parameters.split_infer:
+            split_count = torch.ceil(torch.tensor(signals.shape[1] / self.diar_window_length)).int()
+            sess_emb_vectors, sess_emb_seq, sess_sig_lengths = self.get_range_clus_avg_emb(
+                test_batch, test_data_collection,
+            )
+            with autocast():
+                _preds, scale_weights = self.msdd_model.forward_infer(
+                    input_signal=sess_emb_seq,
+                    input_signal_length=sess_sig_lengths,
+                    emb_vectors=sess_emb_vectors,
+                    targets=None,
+                )
+            _preds = _preds.reshape(len(signal_lengths), split_count * self.diar_window_length, -1)
+            _preds = _preds[:, : signals.shape[1], :]
+        else:
+            with autocast():
+                _preds, scale_weights = self.msdd_model.forward_infer(
+                    input_signal=signals, input_signal_length=signal_lengths, emb_vectors=emb_vectors, targets=_targets
+                )
+        self.max_pred_length = max(_preds.shape[1], self.max_pred_length)
+        preds = torch.zeros(_preds.shape[0], self.max_pred_length, _preds.shape[2])
+        targets = torch.zeros(_preds.shape[0], self.max_pred_length, _preds.shape[2])
+        preds[:, : _preds.shape[1], :] = _preds
+        return preds, targets, signal_lengths
+
+    def run_pairwise_diarization(
+        self, embedding_dir: str = './', device: str = 'cuda'
```

Comment:
embedding_dir and device is not used in this func

Reply:
Removed